File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        n = len(matrix)
        m = len(matrix[0])
        q = deque()
        for i in range(n):
            for j in range(m):
                if matrix[i][j] == 0:
                    q.append((i, j))
        while q:
            x, y = q.popleft()
            for i in range(m):
                matrix[x][i] = 0
            for i in range(n):
                matrix[i][y] = 0
        

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        List<Pair<Integer, Integer>> list = new ArrayList<>();
        int n = matrix.length;
        int m = matrix[0].length;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (matrix[i][j] == 0) {
                    list.add(new Pair(i, j));
                }
            }
        } 
        for (Pair<Integer, Integer> pair : list) {
            for (int i = 0; i < n; i++) {
                matrix[i][pair.getValue()] = 0;
            }
            for (int i = 0; i < m; i++) {
                matrix[pair.getKey()][i] = 0;
            }
        }
    }
}